system/jlib/jevent.cpp

```diff
@@ -1643,65 +1617,61 @@ class CDumpCSVEventVisitor : public CDumpStreamEventVisitor
 
         state = State::Header;
         encodeCSVColumn(markup, "EventName");
-        markup.append(",");
-        encodeCSVColumn(markup, "EventId");
         for(EventAttr a = EventAttr(byte(EvAttrNone) + 1); a < EvAttrMax; a = EventAttr(byte(a) + 1))
         {
-            reverseLookup[queryEventAttributeName(a)] = a;
             markup.append(",");
             encodeCSVColumn(markup, queryEventAttributeName(a));
         }
         dump(true);
         return true;
     }
+
     virtual Continuation visitEvent(EventType id) override
     {
         if (inHeader())
             state = State::Events;
         encodeCSVColumn(markup, queryEventName(id));
-        markup.append(',');
-        StringBuffer tmp;
-        encodeCSVColumn(markup, tmp.append(uint16_t(id)));
-        eventAttrs = headerAttrs;
         return visitContinue;
     }
+
     virtual bool departEvent() override
     {
         for(EventAttr a = EventAttr(byte(EvAttrNone) + 1); a < EvAttrMax; a = EventAttr(byte(a) + 1))
```

Comment:
suggestion: Could using an unsigned a, and avoid the casts.

Reply:
It does look cleaner to remove the casts from the for statements. All casts could be removed if the row and headerAttr datatypes are changed, but I prefer to keep the intent of those types clear by keeping the type. I'll move the casts from the for statements to where needed.